egs/wsj/s5/steps/nnet3/xconfig_to_configs.py

```diff
@@ -170,7 +170,31 @@ def write_config_files(config_dir, all_layers):
             # preserves the backtrace
             raise
 
+    # remove previous init.config
+    try:
+        os.remove(config_dir + '/init.config')
+    except OSError:
+        pass
+
     for basename, lines in config_basename_to_lines.items():
+        # check the lines num before 'output-node':
```

Comment:
You don't have to check the num-lines before 'output-node'. 
Just check the number of times the line starts with 'output-node'.
E.g.:
```
 num_output_node_lines = sum( [ 1 if line.startswith('output-node' ) else 0 
                                                     for line in lines ] )
 ...
```